blocks/identity-block/features/header-account-action/default.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from '@arc-fusion/prop-types';
+import useIdentity from '../../components/Identity';
+
+import './styles.scss';
+
+const HeaderAccountAction = ({ customFields }) => {
+  const { loginURL } = customFields;
+
+  const { Identity, isInitialized } = useIdentity();
+
+  const [loggedIn, setIsLoggedIn] = useState(false);
+  const [user, setUser] = useState(false);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    const isLoggedIn = async () => {
+      setIsLoggedIn(await Identity.isLoggedIn());
+    };
+
+    isLoggedIn();
+  }, [Identity]);
+
+  useEffect(() => {
+    let isActive = true;
+
+    if (loggedIn) {
+      Identity.getUserProfile().then((userProfile) => {
+        if (isActive) {
+          setUser(userProfile);
+        }
+      }).catch((e) => {
+        if (isActive) {
+          setError(e);
+        }
+      });
+    }
+
+    // cancel subscription to useEffect
+    return () => { isActive = false; return null; };
+  }, [Identity, loggedIn]);
+
+  const handleLogout = () => {
+    Identity.logout().then(() => { setIsLoggedIn(false); setUser(null); });
+  };
+
+  if (!isInitialized) {
+    return null;
+  }
+
```

Comment:
@farleyawapo in this logic block, it should say, if logged in, then render that account button